File: Back- End/models/Accommodation.js
const mongoose = require('mongoose');

const AccommodationSchema = new mongoose.Schema({

    district: {
        type: String,
        required: true,
    },
    name: {
        type: String,
        required: true
    },
    description: {
        type: String,
        required: true
    },

    location: {
        type: String,
        required: false
    },
    images: {
        type: Array,
        required: true,
    },
    budget: {
        type: String,
        required: true,
    },
    // tripPersonType: {
    //     type: String,
    //     required: false,
    // },
    // tripType: {
    //     type: String,
    //     required: false,
    // },
    locationLink: {
        type: String,
        required: false,
    },
    contact: {
        type: String,
        required: false,
    },
    dayCost: {
        type: Number,
        required: true,
    },
});

const Accommodation = mongoose.model('Accommodation', AccommodationSchema);
module.exports = Accommodation;

File: Front- End/lib/Models/Accommodation.dart
class Accommodation {
  final String district;
  final String name;
  final String location;
  final String description;
  final List<String> images;
  final String budget;
  final String locationLink;
  final int dayCost;
  final String contact;
  

  Accommodation({
    required this.district,
    required this.name,
    required this.location,
    required this.description,
    required this.images,
    required this.budget,
    required this.locationLink,
    required this.dayCost,
    required this.contact,
  });

  factory Accommodation.fromJson(Map<String, dynamic> json) {
    return Accommodation(
      district: json['district'],
      name: json['name'],
      location: json['location'],
      description: json['description'],
      images: List<String>.from(json['images']),
      budget: json['budget'],
      locationLink: json['locationLink'],
      contact: json['contact'],
      dayCost: json['dayCost'] is String ? int.tryParse(json['dayCost']) ?? 0 : json['dayCost'],
    );
  }
}